client/test/TestFluid.res

```diff
@@ -2802,6 +2787,226 @@ let run = () => {
+    )
+    t(
+      "pressing space completes partial",
+      trueBool,
+      ~pos=4,
+      inputs(list{InsertText("|"), keypress(K.Down), keypress(K.Space)}),
+      "true || ~______",
+    )
+    t(
+      ~expectsPartial=true,
+      "ins | starts partial after variable",
+      aVar,
+      ~pos=8,
+      ins("|"),
+      "variable |~",
+    )
+    t(
+      "pressing pipe twice then space completes partial",
+      trueBool,
+      ~pos=4,
+      inputs(list{InsertText("|"), InsertText("|"), keypress(K.Space)}),
+      "true || ~______",
+    )
+    t(
+      "pressing bs to clear partial reverts for blank rhs",
+      partial("|", or'(anInt, b)),
+      ~pos=7,
+      bs,
+      "12345~",
+    )
+    t(
+      "pressing bs to clear partial reverts for blank rhs, check lhs pos goes to start",
+      partial("|", or'(b, b)),
+      ~pos=8,
+      bs,
+      "~___",
+    )
+    t(
+      "pressing del to clear partial reverts for blank rhs",
+      partial("|", or'(anInt, b)),
+      ~pos=6,
+      del,
+      "12345~",
+    )
+    t(
+      "pressing del to clear partial reverts for blank rhs, check lhs pos goes to start",
+      partial("|", or'(b, b)),
+      ~pos=7,
+      del,
+      "~___",
+    )
+    t(
+      "using bs to remove an infix with a placeholder goes to right place",
+      partial("|", or'(b, b)),
+      ~pos=8,
+      bs,
+      "~___",
+    )
+    t(
+      "using bs to remove an infix with a placeholder goes to right place 2",
+      partial("|", or'(five, b)),
+      ~pos=3,
+      bs,
+      "5~",
+    )
+    t(
+      "deleting binop between bools does not combine them",
+      partial("|", or'(trueBool, falseBool)),
+      ~pos=6,
+      bs,
+      "true~",
+    )
+    t(
+      "pressing bs to clear rightpartial reverts for blank rhs",
+      rightPartial("|", b),
+      ~pos=5,
+      bs,
+      "~___",
+    )
+    t(
+      "pressing bs on a partial over and/or deletes op and combines rhs and lhs",
+      partial("|", or'(anInt, anInt)),
+      ~pos=7,
+      bs,
+      "12345~12345",
+    )
+    t(
+      "using del to remove an infix with a placeholder goes to right place",
+      partial("|", or'(b, b)),
+      ~pos=7,
+      del,
+      "~___",
+    )
+    t(
+      "pressing del to clear rightpartial reverts for blank rhs",
+      rightPartial("|", b),
+      ~pos=4,
+      del,
+      "~___",
+    )
+    t(
+      ~expectsPartial=true,
+      "Pressing pipe while editing a partial works properly",
+      partial("|", or'(trueBool, falseBool)),
+      ~pos=6,
+      ins("|"),
+      "true ||~ false",
+    )
+    t(
+      "changing binop to fn should work",
+      partial("Bool::and", and'(trueBool, falseBool)),
+      ~pos=14,
+      keys(list{K.Enter}),
+      "Bool::and ~true false",
+    )
+    t(
+      "changing fn to binop should work",
+      partial("&&", fn(~mod="Bool", "and", list{trueBool, falseBool})),
+      ~pos=2,
+      keys(list{K.Enter}),
+      "~true && false",
+    )
```

Comment:
not for self to re-review these tests. Not sure how hitting 'enter' is supposed to be converting between fn and binop. (esp in the second case)

Reply:
It completes the partial which is already provided in the test case